LocalNotifications/Sources/LocalNotifications/LocalNotifications.swift

```diff
@@ -0,0 +1,136 @@
+//
+// This code is public domain. You are free to use under the condition
+// you accept that no warranty is given or implied. Use at your own risk.
+//
+import SwiftGodot
+import Foundation
+import UserNotifications
+
+#initSwiftExtension(cdecl: "swift_entry_point", types: [LocalNotifications.self])
+
+@Godot
+class LocalNotifications:RefCounted {
+
+    let HOUR:Int = 60 * 60
+
+    var permitted = false
+    var notificationAccepted = false
+
+    /**
+           Display the local notification permission poup box. Check the value of `permitted` after completed.
+     */
+    @Callable
+    func requestPermission(onComplete:Callable) {
+        Task {
+            let notificationCenter = UNUserNotificationCenter.current()
+            let authorizationOptions: UNAuthorizationOptions = [.alert, .sound, .badge]
+        
+            let options = await notificationCenter.notificationSettings()
+            if options.badgeSetting == .enabled {
+                GD.print("Already Authorised")
+                self.permitted = true
+                onComplete.callDeferred()
+                return
+            }
+
+            do {
+                GD.print("Requesting Authorisation")
+                let authorizationGranted = try await notificationCenter.requestAuthorization(options: authorizationOptions)
+                self.permitted = authorizationGranted
+                onComplete.callDeferred()
+            } catch {
+                GD.print("Authorisation request failed. \(error.localizedDescription)")
+                self.permitted = false
+            }
+        }
+    }
+    
+    @Callable
+    func isPermitted() -> Bool {
+        self.permitted
+    }
+
+    @Callable
+    func wasNotificationAccepted() -> Bool {
+        self.notificationAccepted
+    }
+
+    @Callable
+    func removeAllNotifications() {
+        let notificationCenter = UNUserNotificationCenter.current()
+        let pending = notificationCenter.removeAllPendingNotificationRequests()
+    }
+
+    @Callable
+    func removeNotification(uid:String) {
+        let notificationCenter = UNUserNotificationCenter.current()
+        notificationCenter.removePendingNotificationRequests(withIdentifiers: [uid])
+    }
+
+    /**
+        Request that a notification banner be displayed in the future, as long
+        as the user is not inside the application. `reqeustPermission()` must
+        be called first.
+
+        - parameter uid: Random string or identifier used to delete this notification later if needed
+        - parameter title: title of the local notification.
+        - parameter body: text body of the local notification.
+        - parameter seconds: display notification in how many seconds time.
+        - returns no value is instantly expected. onComplete read the variable `notificationAccepted`
+     */
```

Comment:
Minor but it's always better to use // or /// comments. Because if you are searching through a project you immediately see that a line starts with // whereas that is not the case with /** */ as that comment can start and end anywhere around the line you search for

Most editors have a shortcut to comment/uncomment blocks of code, in VSCodium it's cmd/ctrl + k + c and cmd/ctrl + k + u to comment and uncomment respectively

Reply:
I was just copying the style of a page explaining how to create swift documentation style comments. I found another page from apple on this, so I'll update to to follow https://www.swift.org/documentation/docc/writing-symbol-documentation-in-your-source-files